core/src/main/java/hudson/tools/JDKInstaller.java

```diff
@@ -281,6 +282,51 @@ I tried to locate exactly how InstallShield parses the arguments (and why it use
             fs.delete(logFile);
 
             break;
+
+        case OSX:
+            // Mount the DMG distribution bundle
+            FilePath dmg = parent.createTempDir("jdk", "dmg");
+            exit = launcher.launch()
+                    .cmds("hdiutil", "attach", "-puppetstrings", "-mountpoint", dmg.getRemote(), jdkBundle)
+                    .stdout(log)
+                    .join();
+            if (exit != 0)
+                throw new AbortException(Messages.JDKInstaller_FailedToInstallJDK(exit));
+
+            // expand the installation PKG
+            String installer = dmg.list("*.pkg")[0].getRemote();
+
+            FilePath pkg = parent.createTempDir("jdk", "pkg");
+            pkg.deleteRecursive(); // pkgutil fails if target directory exists
+            exit = launcher.launch()
+                    .cmds("pkgutil", "--expand", installer, pkg.getRemote())
+                    .stdout(log)
+                    .join();
+            if (exit != 0)
+                throw new AbortException(Messages.JDKInstaller_FailedToInstallJDK(exit));
+
+            exit = launcher.launch()
+                    .cmds("umount", dmg.getRemote())
+                    .stdout(log)
+                    .join();
+            if (exit != 0)
+                throw new AbortException(Messages.JDKInstaller_FailedToInstallJDK(exit));
+
+            // We only want the actual JDK sub-package, which "Payload" is actually a tar.gz archive
+            String payload = pkg.list("jdk*.pkg/Payload")[0].getRemote();
+            exit = launcher.launch()
+                    .pwd(parent).cmds("tar", "xzf", payload)
+                    .stdout(log)
+                    .join();
+            if (exit != 0)
+                throw new AbortException(Messages.JDKInstaller_FailedToInstallJDK(exit));
+
+            parent.child("Contents/Home").moveAllChildrenTo(new FilePath(launcher.getChannel(), expectedLocation));
+            parent.child("Contents").deleteRecursive();
+
+            pkg.deleteRecursive();
+            dmg.deleteRecursive();
```

Comment:
I mimic the windows installer here.
Then intention (afaict) is to leave intermediate files for diagnostic purpose